### ISSUE TITLE
Enhancement: Implement advanced semantic similarity methods in RelevanceImpactEvaluator

### ISSUE THREAD

**Issue by JoshuaOliphant** [JoshuaOliphant/ResumeAIAssistant#133]
## Overview
Upgrade the semantic similarity calculation in RelevanceImpactEvaluator from simple word overlap to more sophisticated methods that better capture meaning and context.

## Current Implementation
The current `_calculate_semantic_similarity()` method uses basic set operations:
```python
def _calculate_semantic_similarity(self, text1: str, text2: str) -> float:
    # Simple implementation - could be enhanced with more sophisticated methods
    words1 = set(text1.lower().split())
    words2 = set(text2.lower().split())
    
    intersection = len(words1.intersection(words2))
    union = len(words1.union(words2))
    
    return intersection / union if union > 0 else 0.0
```

## Limitations
- No semantic understanding (synonyms, related concepts)
- Ignores word importance and context
- Sensitive to exact word matching
- No handling of domain-specific terminology relationships

### 1. Sentence Transformers
- Use pre-trained models like SBERT for semantic embeddings
- Calculate cosine similarity between embeddings
- Better capture of semantic meaning
- Example: `all-MiniLM-L6-v2` for efficiency

### 2. TF-IDF with Cosine Similarity
- Weight words by importance in corpus
- Reduce impact of common words
- Better differentiation of relevant content
- Can combine with word embeddings

### 3. Word Embeddings (Word2Vec/GloVe)
- Map words to vector space
- Calculate similarity based on vector distances
- Handle synonyms and related terms
- Domain-specific fine-tuning possible

### 4. Hybrid Approach
- Combine multiple methods with weighted scoring
- Use fast methods first, advanced for close matches
- Configurable based on accuracy/performance needs

## Implementation Example
```python
def _calculate_semantic_similarity(self, text1: str, text2: str) -> float:
    # Option 1: Sentence transformers
    embeddings1 = self.model.encode(text1)
    embeddings2 = self.model.encode(text2)
    similarity = cosine_similarity(embeddings1, embeddings2)
    
    # Option 2: TF-IDF
    vectorizer = TfidfVectorizer()
    vectors = vectorizer.fit_transform([text1, text2])
    similarity = cosine_similarity(vectors[0], vectors[1])[0][0]
    
    return similarity
```

## Benefits
- More accurate similarity measurements
- Better handling of paraphrasing and synonyms
- Context-aware comparisons
- Improved match score calculations
- Better optimization targeting

## Performance Considerations
- Model loading time (cache models)
- Processing time for embeddings
- Memory usage for vector storage
- Batch processing opportunities

## Related Files
- `evaluation/evaluators/quality.py` (lines 316-325)
- Used in multiple methods for score calculation

## Dependencies
Potential additions:
- `sentence-transformers`
- `scikit-learn` (already in project)
- `numpy` (already in project)
- Pre-trained models

**Priority**: Medium-High
**Estimated Effort**: 2-3 days
**Related to**: #114
